### ISSUE TITLE
mixed content message caused by analytics callback

### ISSUE THREAD

**Issue by childers** [GMOD/jbrowse#660]
We're running all of our traffic through https, and are seeing mixed content messages due to the built in analytics using an http connection. Can you enable support for your analytics callback to use https if appropriate? I'd rather our users not see warnings if we can avoid it.

http://jbrowse.org/analytics/clientReport.....

**Comment by enuggetry** [comment]
I don't have an https config, so you'll have to tell me if it works.

**Comment by cmdcolin** [comment]
Not sure if jbrowse is running https

Simply going to the clientReport times out under https

You can also disable the analytics if you want

**Comment by enuggetry** [comment]
good point. I'll add a cert to the server.

**Comment by cmdcolin** [comment]
If you do get that setup then you can just send the request straight to https instead of conditioning on window.location.protocol.

It gets confused running on electron because then it uses file:/// as protocol and then electron won't get analytics ;)

**Comment by enuggetry** [comment]
If we condition that request to use http for the electron app, will it
still throw a mixed content message?

**Comment by cmdcolin** [comment]
the best scenario is to just always use https for requests, i.e. you could hardcode the https

requesting https via a http page is fine
requesting https via the electron app is fine

the only scenario that is bad is requesting an http page from an https source, which is what gives this mixed content message the childers sees.

If we can't setup the https in time, I would probably revert the window.location.protocol change and then suggest @childers disable analytics for his instance to avoid message

**Comment by enuggetry** [comment]
I'll finish that off today.

**Comment by terrymun** [comment]
I am using jBrowse over a secure channel, and have to change line 1621 in `src/Browser.js` to force it to honour the `document.location.protocol` instead of listening on the `clientReport` object to get it to use `https://`.

I have replaced the following lines:

```javascript
    var protocol = "https";

    // overridable protocol
    if (typeof this.config.clientReport != "undefined" && typeof this.config.clientReport.protocol != "undefined")
        protocol = this.config.clientReport.protocol;

    // phone home with a GET request made by a script tag
    var clientReport = protocol + '://jbrowse.org/analytics/clientReport?'
               + dojo.objectToQuery( stats );
```

With this:

```javascript
    var clientReport = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '://jbrowse.org/analytics/clientReport?'
               + dojo.objectToQuery( stats );
```

**Comment by enuggetry** [comment]
Thanks.

I had configured the server to accept https, was it still not working?

**Comment by terrymun** [comment]
> I had configured the server to accept https, was it still not working?

Did I miss a configuration file when setting up jBrowse? Otherwise, in my honest opinion, I'd rather have the document's location protocol dictate the protocol over which external resources are loaded, instead of an implicit configuration.

**Comment by enuggetry** [comment]
Inheriting the protocol is probably a better approach.  Thanks.

**Comment by vivekkrish** [comment]
Could it just be a protocol relative URL? Like so:
```
var clientReport = '//jbrowse.org/analytics/clientReport?' + dojo.objectToQuery( stats );
```
Our does that have compatibility issues with particular browsers (and versions)?

**Comment by cmdcolin** [comment]
@vivekkrish I just tested that and it ends up loading file://jbrowse.org/ in the desktop version, so that's the holdout for this.

also, this is sort of just gossip, but you can see here, that protocol relative urls are not really necessary. https://twitter.com/paul_irish/status/588502455530311680?lang=en

we can use https everywhere (which is what current code does!)

**Comment by cmdcolin** [comment]
I believe this issue was fixed in 1.12.0 :)

If we have any continued issues maybe we can have a new bug

**Comment by enuggetry** [comment]
Thanks, Colin.
Yes, we added an SSL cert to our analytics server to accept https.  It would work for both https and http, now.
